smoke-tests/images/servlet/build.gradle

```diff
@@ -96,11 +102,16 @@ createDockerTasks(buildLinuxTestImagesTask, targets, false)
 createDockerTasks(buildWindowsTestImagesTask, targets, true)
 
 def configureImage(Task parentTask, server, dockerfile, version, vm, jdk, warProject, Map<String, String> extraArgs, boolean isWindows, String extraTag) {
-  // Using separate build directory for different war files allows using the same app.war filename
-  def dockerWorkingDir = new File(project.buildDir, "docker-$warProject")
+  // Using separate build directory for different images
+  def dockerWorkingDir = new File(project.buildDir, "docker-$server-$version-jdk$jdk-$vm-$warProject")
   def dockerFileName = isWindows ? "${dockerfile}.windows.dockerfile" : "${dockerfile}.dockerfile"
   def platformSuffix = isWindows ? "-windows" : ""
 
+  if (!new File("src/${dockerFileName}").exists()) {
+    println "Skipping ${dockerFileName} as it doesn't exist"
+    return
+  }
```

Comment:
I liked that you could look at the matrix above to see what combinations are run, but that is no guaranteed with this. maybe a flag in the matrix not to run on windows (or something like that)?